docs/deploying.adoc

```diff
@@ -9,11 +9,242 @@ Put the filter jars in some directory
+    targetCluster:
+      bootstrap_servers: myprivatecluster:9092  # <2>
+    clusterNetworkAddressConfigProvider:
+      type: PortPerBroker                       # <3>
+      config:
+        bootstrapAddress: mypublickroxylicious:9192    # <4>
+----
+<1> The name of the virtual cluster.
+<2> The bootstrap of the (physical) Kafka Cluster.  This is the Kafka Cluster being proxied.
+<3> The name of a cluster network address config provider. The built-in types are `PortPerBroker` and `SniRouting`.
+<4> The hostname and port of the bootstrap that will be used by the Kafka Clients.  The hostname must be resolved
+by the clients.
+
+This configuration declares a virtual cluster called `demo`.  The physical Kafka Cluster being proxied is the defined
+by the `targetCluster` element.  In this example, the `PortPerBroker` scheme is used by Kroxlicious to present the
+virtual cluster to the clients.  Under this schema, Kroxylicious will open a port for each broker of the target cluster
+with port numbers beginning at `9192` +1.  So, if the target cluster has three brokers, Kroxylicious will bind 9192 for
+bootstrap and 9193-9195 inclusive to allow the clients to connect to each broker.
+
+=== Cluster Network Address Config Providers
+
+The Cluster Network Address Config Provider controls how the virtual cluster is presented oo the network. Two
+alternatives are supported: `PortPerBroker` and `SniRouting` which have different characteristics which make each
+suitable for different use-cases. They are described next.
+
+==== PortPerBroker scheme
+
+In the `PortPerBroker` scheme, Kroxylicious automatically opens port for each virtual cluster's bootstrap and
+one port per broker of each target cluster.  So, if you have two virtual clusters, each targeting a Kafka Cluster
+of three brokers, Kroxylicious will bind eight ports in total.
+
+The `PortPerBroker` scheme can be used with either upstream plain or upstream TLS connections.
+
+[source, yaml]
+----
+clusterNetworkAddressConfigProvider:
+  type: PortPerBroker
+  config:
+    bootstrapAddress: mycluster.kafka.com:9192                                 # <1>
+    brokerAddressPattern: mybroker-$(nodeId).mycluster.kafka.com:$(portNumber) # <2>
+    brokerStartPort: 9193                                                      # <3>
+    numberOfBrokerPorts: 3                                                     # <4>
+    bindAddress: 192.168.0.1                                                   # <5>
+----
+<1> The hostname and port of the bootstrap that will be used by the Kafka Clients.
+<2> (Optional) The broker address pattern used to form the broker addresses.  If not defined, it defaults to the
+hostname part of the `bootstrapAddress` and the port number allocated to the broker.
+<3> (Optional) The starting number for broker port range. Defaults to the port of the `bootstrapAddress` plus 1.
+<4> (Optional) The maximum number of brokers of ports that will be defined from the range.  Defaults to 3.
```

Comment:
We could note that it can be greater than the current cluster size

Reply:
How does that look?

Reply:
Looks good